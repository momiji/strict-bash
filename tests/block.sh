{
    
}
{}
{true}
{1;2}

{true} || exit
