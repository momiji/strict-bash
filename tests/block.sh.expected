start: State(name='start', start=0, end=0, depth=0)
    start: State(name='statement', start=0, end=0, depth=1)
        start: State(name='{', start=0, end=0, depth=2)
            start: State(name='statement', start=2, end=2, depth=3)
            end: State(name='statement', start=2, end=6, depth=3) => '    '
        end: State(name='{', start=0, end=8, depth=2) => '{\n    \n}'
    end: State(name='statement', start=0, end=8, depth=1) => '{\n    \n}'
    start: State(name='statement', start=9, end=9, depth=1)
        start: State(name='{', start=9, end=9, depth=2)
        end: State(name='{', start=9, end=11, depth=2) => '{}'
    end: State(name='statement', start=9, end=11, depth=1) => '{}'
    start: State(name='statement', start=12, end=12, depth=1)
        start: State(name='{', start=12, end=12, depth=2)
            start: State(name='statement', start=13, end=13, depth=3)
                start: State(name='command', start=13, end=13, depth=4)
                    start: State(name='arg', start=13, end=13, depth=5)
                    end: State(name='arg', start=13, end=17, depth=5) => 'true'
                end: State(name='command', start=13, end=17, depth=4) => 'true'
            end: State(name='statement', start=13, end=17, depth=3) => 'true'
        end: State(name='{', start=12, end=18, depth=2) => '{true}'
    end: State(name='statement', start=12, end=18, depth=1) => '{true}'
    start: State(name='statement', start=19, end=19, depth=1)
        start: State(name='{', start=19, end=19, depth=2)
            start: State(name='statement', start=20, end=20, depth=3)
                start: State(name='command', start=20, end=20, depth=4)
                    start: State(name='arg', start=20, end=20, depth=5)
                    end: State(name='arg', start=20, end=21, depth=5) => '1'
                end: State(name='command', start=20, end=21, depth=4) => '1'
            end: State(name='statement', start=20, end=21, depth=3) => '1'
            start: State(name='statement', start=22, end=22, depth=3)
                start: State(name='command', start=22, end=22, depth=4)
                    start: State(name='arg', start=22, end=22, depth=5)
                    end: State(name='arg', start=22, end=23, depth=5) => '2'
                end: State(name='command', start=22, end=23, depth=4) => '2'
            end: State(name='statement', start=22, end=23, depth=3) => '2'
        end: State(name='{', start=19, end=24, depth=2) => '{1;2}'
    end: State(name='statement', start=19, end=24, depth=1) => '{1;2}'
    start: State(name='statement', start=26, end=26, depth=1)
        start: State(name='{', start=26, end=26, depth=2)
            start: State(name='statement', start=27, end=27, depth=3)
                start: State(name='command', start=27, end=27, depth=4)
                    start: State(name='arg', start=27, end=27, depth=5)
                    end: State(name='arg', start=27, end=31, depth=5) => 'true'
                end: State(name='command', start=27, end=31, depth=4) => 'true'
            end: State(name='statement', start=27, end=31, depth=3) => 'true'
        end: State(name='{', start=26, end=32, depth=2) => '{true}'
        start: State(name='operator', start=33, end=33, depth=2)
        end: State(name='operator', start=33, end=35, depth=2) => '||'
        start: State(name='command', start=36, end=36, depth=2)
            start: State(name='arg', start=36, end=36, depth=3)
            end: State(name='arg', start=36, end=40, depth=3) => 'exit'
        end: State(name='command', start=36, end=40, depth=2) => 'exit'
    end: State(name='statement', start=26, end=40, depth=1) => '{true} || exit'
end: State(name='start', start=0, end=41, depth=0) => '{\n    \n}\n{}\n{true}\n{1;2}\n\n{true} || exit\n'
